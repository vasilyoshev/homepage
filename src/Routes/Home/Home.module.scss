.gridContainer {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(320px, 1fr));
  grid-auto-flow: dense;
  grid-gap: 0.2em;
  align-items: start;
  margin: auto;
}
