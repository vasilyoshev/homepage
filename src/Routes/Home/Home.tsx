import React from 'react';
import { useSelector } from 'react-redux';
import { AddWidgetFab } from 'components';
import { CalendarWidget, TodoWidget, CryptoCurrencyTrackerWidget } from 'widgets';
import { selectIsCalendarWidgetActive, selectIsTodoWidgetActive, selectIsCryptoCurrencyWidgetActive } from 'slices';

export const Home: React.FC = () => {
  const isCalendarWidgetActive = useSelector(selectIsCalendarWidgetActive);
  const isTodoWidgetActive = useSelector(selectIsTodoWidgetActive);
  const isCryptoCurrencyWidgetActive = useSelector(selectIsCryptoCurrencyWidgetActive);

  return (
    <>
      <AddWidgetFab />
      {isCryptoCurrencyWidgetActive && <CryptoCurrencyTrackerWidget />}
      {isCalendarWidgetActive && <CalendarWidget />}
      {isTodoWidgetActive && <TodoWidget />}
    </>
  );
};
