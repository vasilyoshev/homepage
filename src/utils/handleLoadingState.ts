import axios from 'axios';
import { decrementLoadingCount, incrementLoadingCount } from 'slices';
import store from 'store';

export const handleLoadingState = (): void => {
  axios.interceptors.response.use(
    (response) => {
      setTimeout(() => store.dispatch(decrementLoadingCount()), 200);

      return response;
    },
    () => {
      store.dispatch(decrementLoadingCount());
    },
  );

  axios.interceptors.request.use((request) => {
    store.dispatch(incrementLoadingCount());

    return request;
  });
};
