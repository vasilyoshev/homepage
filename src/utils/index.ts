export * from './handleLoadingState';
export * from './testUtils';
