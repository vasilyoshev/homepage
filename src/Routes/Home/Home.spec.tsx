import { Home } from './Home';
import { render } from 'utils';

describe('Home', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<Home />);

    expect(asFragment()).toMatchSnapshot();
  });
});
