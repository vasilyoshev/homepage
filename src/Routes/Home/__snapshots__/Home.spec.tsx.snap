// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders successfully 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiFab-root addWidgetFab MuiFab-primary"
    tabindex="0"
    title="Add Widget"
    type="button"
  >
    <span
      class="MuiFab-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <div
    class="coinApp"
  >
    <div
      class="coinSearch"
    >
      <h1
        class="coinText"
      >
        Search a currency
      </h1>
      <form>
        <input
          class="coinInput"
          placeholder="Search"
          type="text"
        />
      </form>
    </div>
    <div
      class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <table
        aria-label="a dense table"
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
              scope="col"
            >
              Name
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
              scope="col"
            >
              Logo
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
              scope="col"
            >
              Symbol
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
              scope="col"
            >
              Price
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
              scope="col"
            >
              Price change in 24h
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
              scope="col"
            >
              Market capitalization
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        />
      </table>
    </div>
  </div>
</DocumentFragment>
`;
